CRM/Speakcivi/Logic/Activity.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+class CRM_Speakcivi_Logic_Activity {
+
+  /**
+   * Get activity type id. If type isn't exist, create it.
+   *
+   * @param string $activityName
+   *
+   * @return int
+   * @throws \CiviCRM_API3_Exception
+   */
+  private static function getTypeId($activityName) {
+    $params = array(
```

Comment:
it costs nothing and is good practice to store the type in a static variable so you don't fetch everytime ( if $type) then return type; then fetch